pkg/controller/machine/controller.go

```diff
@@ -194,9 +194,15 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 }
 
 func (r *ReconcileMachine) getCluster(ctx context.Context, machine *clusterv1.Machine) (*clusterv1.Cluster, error) {
+	if machine.Labels["cluster"] == "" {
+		klog.Warningf("Machine %q in namespace %q has no cluster label, returning nil", machine.Name, machine.Namespace)
+		return nil, nil
```

Comment:
shouldn't this return an error, instead of `nil` for error?

Reply:
See https://github.com/kubernetes-sigs/cluster-api/pull/728#discussion_r254057817, I think it makes sense to allow Machines function without a cluster

Reply:
This should probably be just info rather than a warning, see same discussion https://github.com/kubernetes-sigs/cluster-api/pull/644#discussion_r242743621

Reply:
@vincepri I am not sure I understand what "allow machines to function without a cluster means". If it means that the machines should be created/updated/deleted, then we should return and should allow the controller to perform the reconciliation. Correct me if I am wrong.